tests/TestCase/Cache/Engine/RedisEngineTest.php

```diff
@@ -127,6 +130,102 @@ public function testConfigDsn(): void
             'host' => 'localhost',
             'scheme' => 'redis',
             'scanCount' => 10,
+            'context' => [],
+        ];
+        $this->assertEquals($expecting, $config);
+
+        Cache::drop('redis_dsn');
+    }
+
+    /**
+     * testConfigDsnSSLContext method
+     */
+    public function testConfigDsnSSLContext(): void
+    {
+        $url = 'redis://localhost:' . $this->port;
+
+        $url .= '?ssl_ca=/tmp/cert.crt';
+        $url .= '&ssl_key=/tmp/local.key';
+        $url .= '&ssl_cert=/tmp/local.crt';
+
+        Cache::setConfig('redis_dsn', compact('url'));
+
+        $config = Cache::pool('redis_dsn')->getConfig();
+        $expecting = [
+            'prefix' => 'cake_',
+            'duration' => 3600,
+            'groups' => [],
+            'server' => 'localhost',
+            'port' => $this->port,
+            'tls' => false,
+            'timeout' => 0,
+            'persistent' => true,
+            'password' => false,
+            'database' => 0,
+            'unix_socket' => false,
+            'host' => 'localhost',
+            'scheme' => 'redis',
+            'scanCount' => 10,
+            'context' => [
+                'ssl' => [
+                    'cafile' => '/tmp/cert.crt',
+                    'local_cert' => '/tmp/local.crt',
+                    'local_pk' => '/tmp/local.key',
+                ],
+            ],
+            'ssl_ca' => '/tmp/cert.crt',
+            'ssl_key' => '/tmp/local.key',
+            'ssl_cert' => '/tmp/local.crt',
```

Comment:
Having users change the top-level keys but the adapter using the nested one is my concern. We need to give users a single interface to work with.